news-bundle/src/Resources/contao/dca/tl_news.php

```diff
@@ -62,7 +62,7 @@
 			(
 				'id' => 'primary',
 				'alias' => 'index',
-				'pid,start,stop,published' => 'index'
+				'pid,featured,published,start,stop' => 'index'
```

Comment:
Many news queries don’t ask for the `featured` column, how about creating and index for both variants?
```suggestion
				'pid,featured,published,start,stop' => 'index',
				'pid,published,start,stop' => 'index'
```

Reply:
On my system, the composite index is also used if the `featured` column is not part of the query.

Reply:
But it can only be used for the `pid` then, not for `published`.
We should at least change the order to `pid,published,featured,start,stop`

Reply:
Are you sure?

<img width="1071" alt="" src="https://user-images.githubusercontent.com/1192057/92141830-45bc9e80-ee13-11ea-80d3-dd4ad5f505ae.png">

Reply:
Does key_len increase when you change the order?

Reply:
> On my system, the composite index is also used if the featured column is not part of the query.

It should show "Using index condition; Using where" because it's using the index for the `pid` part. Are you sure the column isn't truncated in your output?

Reply:
> Does key_len increase when you change the order?

Which order are you referring to?

Reply:
The composite index order: `pid,published,featured,start,stop`

Reply:
It does indeed.

<img width="1025" alt="" src="https://user-images.githubusercontent.com/1192057/92212204-eb165780-ee91-11ea-8e60-7b99e777ab56.png">

Reply:
However, now that both indexes exist, MySQL still uses the one without the `featured` column even when the query contains it. 🤔 

<img width="1109" alt="" src="https://user-images.githubusercontent.com/1192057/92212538-6a0b9000-ee92-11ea-875f-71d6ef2de601.png">

<img width="781" alt="" src="https://user-images.githubusercontent.com/1192057/92212625-89a2b880-ee92-11ea-9456-3befc84ff794.png">

Reply:
Ah, this seems to be because I queried for `featured=''`. MySQL uses the other index when I query for `featured='1'`.

<img width="1168" alt="" src="https://user-images.githubusercontent.com/1192057/92212724-b3f47600-ee92-11ea-943a-31952270fef7.png">

So the second index only covers the case that there is a news module that shows only featured news. No idea if that is worth the extra index?

Reply:
I think we could then just use one index with the “better” order: `pid,published,featured,start,stop`

Reply:
Changed in 7be1eaeb15fba0fc28b0d399b74597ba7574a612.